proposals/vprim/README.md

```diff
@@ -0,0 +1,121 @@
+
+Let's use points as a motivating example. Consider an OpenUSD file that contains a mesh reconstruction of data from a point cloud scan. A user might benefit from visualizing the points on top of the mesh to aid in debugging some errors in the reconstruction. The user might choose the `Points` schema to encode positions. And they might be able to hide the source point cloud from rendering contexts using `purpose` tagging and/or variant sets. But they will be unable to make the points all the same size in camera space in a specification compliant renderer. Preview optimizations that might give a user their desired effect like those found in Storm exist primarily for performance reasons and may not persist across renderers, sessions, and future releases.
+
+A visualization focused specification of points could look something like this--
+
+```
+def VizPoints "PointCloud" {
+    point3d[] points = [...]
+    float[] pixels = [...]
+    color3f[] primvars:displayColor = [(1, 0, 0)]
+}
+```
+
+`VizPoints` could be rendered in camera but may also be composited against a depth buffer from a physically based renderer.
+
+### Styling
+`Vprims` still support primvars and would be primarily styled using `displayColor` and `displayOpacity`.
+
+We'd like to leave open the possibility of a `viz` material purpose as a holder for describing nondiegetic styles best described through material network, but this proposal doesn't have a concrete definition of this yet.
+
+### API Schema and Grouping Adapters
+This proposal frames screen space points (and paths) visualization primitives as fundamentally different from their `Gprim` counterparts, warranting their own first class `Vprim` typed schemas. Does this proposal set the precedent that other `Gprim` types like `Cube` or `Cylinder` require visualization equivalents if they wanted to be adapted into the "visualization" domain? We'd suggest no and offer two solutions that have prior art in OpenUSD.
+
+If a primitive retains its underlying definition, `UsdLux` already provides a solution to the problem of adapting `Gprim`s into another domain. Rather than a `MeshLight`, the `Mesh` `Gprim` is adapted via an API schema. `Gprim`s could be similarly "adapted" into the visualization domain. To avoid "schema slicing" an adaptation must not change the underlying definition or require their own distinct `purpose` or `visibility` tagging. Consideration must be also given to if or how these schemas affect interactive picking. A non-slicing visualization effect might include sillhoutte outlines, glows, or overlays applied to a `Gprim` to style or highlight its importance.
```

Comment:
I guess there's a philosophical question; practically, we'll implement overlay and silhouette as image-plane operations because we don't want them to participate in rendering and after rendering is done image-plane is the best way to express things.  Any kind of image operation can reasonably expect to get a certain amount of data, e.g. color, depth, id, etc if needed.  I think silhouette can be done as an image operation as an edge-detect convolution on primId?  And overlay can of course be done trivially as a primId match.

Internally we do some more complicated things where we render additional 3d geometry with custom compositing, e.g. drawing rotation widgets on top as transparent if they fail the depth test and as opaque if they pass the depth test.  I don't know if we want to try to model that stuff here.  In games that's something that can be modelled by materials, but since that kind of thing isn't physically based it's not captured by UsdShade.  It's certainly useful for those kinds of prims to identify that they don't participate in light transport, at least, and I could imagine having hydra support for rendering 3d prims into the scene after the primary render (e.g. by running Viz prims through Storm and comping them in).

Reply:
Everything you said makes sense to me Tom-- I'd expect that most styles would be implemented as image shaders that way.

I suspect there's probably going to eventually be a need for shaders to support more advanced effects, but I also suspect that there's a lot of concepts that benefit from a simplified initial specification.